src/extension.ts

```diff
@@ -34,7 +34,9 @@ export function activate(context: vscode.ExtensionContext): void {
     // Couldn't load keytar, proceed without storing & loading connections.
   }
 
-  mdbExtension = new MDBExtensionController(context);
+  const needTelemetry = true;
+
+  mdbExtension = new MDBExtensionController(context, needTelemetry);
```

Comment:
Probably you already considered it, but we could do constructor injection here and pass the telemetry as a dependency instead:

```
new MDBExtensionController(context, telemetryController)
```

and use `sinon.createStubInstance` to mock the telemetry away with all the functions and events, and even spy to test that telemetry events will be actually sent.

For segment specifically another alternative is to use a fake or empty segment key. Last time i checked the segment library was not exploding and just not sending events.

Reply:
This is an entry point, we don't have controllers here yet.